frontend/src/lib/components/common/button/Button.svelte

```diff
@@ -1,79 +1,114 @@
-	export let color: 'blue' | 'dark' | 'light' = 'blue'
-	export let variant: 'contained' | 'border' = 'contained'
+	export let size: Button.Size = 'md'
+	export let color: Button.Color = 'blue'
+	export let variant: Button.Variant = 'contained'
 	export let btnClasses: string = ''
 	export let disabled: boolean = false
 	export let href: string | undefined = undefined
-	export let target: '_self' | '_blank' = '_self'
+	export let target: Button.Target = '_self'
 
 	export let startIcon: { icon: any; classes?: string } | undefined = undefined
 	export let endIcon: { icon: any; classes?: string } | undefined = undefined
 
-	function getColorClasses() {
-		switch (color) {
-			case 'blue':
-				return 'bg-blue-500 hover:bg-blue-700 focus:ring-blue-300 text-white '
-			case 'light':
-				return 'text-gray-800 bg-white hover:bg-gray-100 focus:ring-gray-300'
-			default:
-				return ''
+	const dispatch = createEventDispatcher()
+
+	// Order of classes: border, border modifier, bg, bg modifier, text, text modifier, everything else
+	const colorVariants: Record<Button.Color, Record<Button.Variant, string>> = {
+		blue: {
+			border:
+				'border-blue-500 hover:border-blue-700 bg-white hover:bg-blue-100 text-blue-500 hover:text-blue-700 focus:ring-blue-300',
+			contained: 'bg-blue-500 hover:bg-blue-700 text-white focus:ring-blue-300'
+		},
+		dark: {
+			border:
+				'border-gray-800 hover:border-gray-900 bg-white hover:bg-gray-200 text-gray-800 hover:text-gray-900 focus:ring-gray-300',
+			contained: 'bg-gray-700 hover:bg-gray-900 text-white focus:ring-gray-300'
+		},
+		light: {
+			border:
+				'border-gray-700 hover:border-gray-800 bg-white hover:bg-gray-100 text-gray-700 hover:text-gray-800 focus:ring-gray-300',
+			contained: 'bg-white hover:bg-gray-100 text-gray-700 focus:ring-gray-300'
 		}
 	}
 
-	function getSizeClasses() {
-		switch (size) {
-			case 'xs':
-				return 'text-xs'
-			case 'sm':
-				return 'text-sm'
-			case 'md':
-				return 'text-md'
-			case 'lg':
-				return 'text-lg'
-			case 'xl':
-				return 'text-xl'
-			default:
-				return ''
-		}
+	const sizeClass: Record<Button.Size, string> = {
+		xs: 'text-xs px-2.5 py-1',
+		sm: 'text-sm px-2.5 py-1',
+		md: 'text-md px-3 py-1',
+		lg: 'text-lg px-4 py-2',
+		xl: 'text-xl px-5 py-2'
+	}
+
+	const iconScale: Record<Button.Size, number> = {
+		xs: 0.7,
+		sm: 0.8,
+		md: 1,
+		lg: 1.1,
+		xl: 1.2
 	}
 
 	const buttonProps = {
 		class: classNames(
-			getColorClasses(),
-			getSizeClasses(),
+			colorVariants[color][variant],
+			variant === 'border' ? 'border' : '',
+			sizeClass[size],
 			'focus:ring-4 font-medium',
-			'px-4 py-2',
 			'rounded-md',
-			'flex justify-center items-center',
-			variant === 'border' ? 'border' : '',
-			btnClasses
+			'flex justify-center items-center text-center',
+			btnClasses,
+			disabled ? 'pointer-events-none cursor-default filter grayscale' : ''
 		),
 		disabled
 	}
+
+	const onClick = () => dispatch('click')
```

Comment:
The previous implementation worked using: https://svelte.dev/tutorial/dom-event-forwarding
Did you have any issues ?

Reply:
I didn't know about this, it's now reverted.